internal/querynodev2/delegator/delegator.go

```diff
@@ -452,30 +504,36 @@ func executeSubTasks[T any, R interface {
 				case errCh <- err: // must be the first
 				default: // skip other errors
 				}
-				cancel()
 				return
 			}
 			resultCh <- result
 		}(task)
 	}
 
-	wg.Wait()
-	close(resultCh)
-	select {
-	case err := <-errCh:
-		log.Warn("Delegator execute subTask failed",
-			zap.String("taskType", taskType),
-			zap.Error(err),
-		)
-		return nil, err
-	default:
-	}
-
 	results := make([]R, 0, len(tasks))
-	for result := range resultCh {
-		results = append(results, result)
+	for i := 0; i < len(tasks); i++ {
```

Comment:
this won't work well if the error occurs at the last task

Reply:
if err happens at the last task, the following select can only end up in `errCh` branch, which is totally fine